pkg/cloudprovider/providers/gce/gce_instances.go

```diff
@@ -90,9 +90,19 @@ func (gce *GCECloud) NodeAddresses(_ context.Context, _ types.NodeName) ([]v1.No
 	if err != nil {
 		return nil, fmt.Errorf("couldn't get external IP: %v", err)
 	}
+	internalDNSFull, err := metadata.Get("instance/hostname")
```

Comment:
is it safer to log and ignore errors getting the DNS names so the IPs can still be populated?

Reply:
Done. I'm not entirely sure about the guarantees of these metadata fields being present.